hadoop-ozone/insight/src/main/java/org/apache/hadoop/ozone/insight/datanode/RatisInsight.java

```diff
@@ -39,31 +39,43 @@
  */
 public class RatisInsight extends BaseInsightPoint implements InsightPoint {
 
+  public static final String PIPLINE_FILTER = "pipeline";
   private OzoneConfiguration conf;
 
   public RatisInsight(OzoneConfiguration conf) {
     this.conf = conf;
   }
 
   @Override
-  public List<LoggerSource> getRelatedLoggers(boolean verbose) {
+  public List<LoggerSource> getRelatedLoggers(boolean verbose,
+      Map<String, String> filters) {
+    if (filters == null || !filters.containsKey(PIPLINE_FILTER)) {
+      throw new IllegalArgumentException(PIPLINE_FILTER
+          + " filter should be specified (-f pipline=<pipelineid)");
+    }
+
+    String pipelineId = filters.get(PIPLINE_FILTER);
     List<LoggerSource> result = new ArrayList<>();
-    try {
-      Optional<Pipeline> pipeline;
-      try (ScmClient scmClient = createScmClient(conf)) {
-        pipeline = scmClient.listPipelines()
-            .stream()
-            .filter(d -> d.getNodes().size() > 1)
-            .findFirst();
+
+    try (ScmClient scmClient = createScmClient(conf)) {
+      Optional<Pipeline> pipelineSelection = scmClient.listPipelines()
+          .stream()
+          .filter(d -> d.getNodes().size() > 1)
```

Comment:
Do we need the size filter, given that pipeline ID is required?

Reply:
Not sure if this insight is useful for Ratis/ONE but let's remove it. If we would like to forbid the usage for Ratis/One it should be done with a more explicit check.

Line is removed.